yjit/src/cruby.rs

```diff
@@ -735,3 +735,42 @@ mod manual_defs {
     pub const RUBY_OFFSET_ICE_VALUE: i32 = 8;
 }
 pub use manual_defs::*;
+
+/// Interned ID values for Ruby symbols and method names.
+/// See [crate::cruby::ID] and usages outside of YJIT.
+pub(crate) mod ids {
+    use std::sync::atomic::AtomicU64;
+    macro_rules! def_ids {
+        ($(name: $ident:ident content: $str:literal)*) => {
+            $(pub static $ident: AtomicU64 = AtomicU64::new(0);)*
+
+            pub(crate) fn init() {
+                $(
+                    let content = &$str;
+                    let ptr: *const u8 = content.as_ptr();
+
+                    $ident.store(
+                        unsafe { $crate::cruby::rb_intern2(ptr.cast(), content.len() as _) },
+                        std::sync::atomic::Ordering::Relaxed
+                    );
+                )*
+
+            }
+        }
+    }
+
+    def_ids! {
+        name: NULL               content: b""
+        name: min                content: b"min"
+        name: max                content: b"max"
+        name: hash               content: b"hash"
+        name: respond_to_missing content: b"respond_to_missing?"
+    }
+}
+
+macro_rules! rb_id {
```

Comment:
WDYT about calling it `ID!`? Would that be too confusing? `rb_id` feels like a C API of CRuby, but it doesn't exist in the interpreter code. 

I personally feel it's easier to understand it's an `ID` when it's `ID!(min)` instead of `rb_id!(min)`.

Reply:
Maybe `RUBY_ID!()` ? Just because ID is kind of a generic concept.

Reply:
If we want to have a prefix, I would prefer `RB_` over `RUBY_` since `RUBY_` seems too long compared to the main `ID`, `RB_` prefix is also common enough. So, I'd vote for `RB_ID!()` (it still seems better than `rb_id!()` as to the point I made above).

Reply:
I like `ID!` too. The macro being nested under `cruby` probably makes it specific enough, and LSP hover shows the full path for anyone who wants to dig. It matching the C type name feels nice.